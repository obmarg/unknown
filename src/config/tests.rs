use std::{fs::File, io::Read};

use super::*;

#[test]
fn test_can_load_project_file() {
    let mut str_data = String::new();
    File::open("config-examples/project.kdl")
        .unwrap()
        .read_to_string(&mut str_data);

    insta::assert_debug_snapshot!(
        parse_project_file(&PathBuf::from("blah/project.kdl"), &str_data)
            .map_err(|e| miette::Report::new(e.0))
    )
}

#[test]
fn test_can_load_workspace_file() {
    let mut str_data = String::new();
    File::open("config-examples/workspace.kdl")
        .unwrap()
        .read_to_string(&mut str_data);

    insta::assert_debug_snapshot!(parse_workspace_file(
        &PathBuf::from("blah/workspace.kdl"),
        &str_data
    )
    .map_err(|e| miette::Report::new(e.0)))
}
