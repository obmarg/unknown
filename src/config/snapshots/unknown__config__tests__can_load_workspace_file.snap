---
source: src/config/tests.rs
assertion_line: 22
expression: "workspace_from_str(&str_data).map_err(|e| miette::Report::new(e.0))"
---
Ok(
    WorkspaceFile {
        name: "my-workspace",
        project_paths: [
            "**",
        ],
    },
)
